go/grpcweb/websocket_wrapper.go

```diff
@@ -10,25 +10,55 @@ import (
 	"net/http"
 	"net/textproto"
 	"strings"
+	"time"
 
 	"github.com/gorilla/websocket"
 	"golang.org/x/net/http2"
 )
 
 type webSocketResponseWriter struct {
-	writtenHeaders bool
-	wsConn         *websocket.Conn
-	headers        http.Header
-	flushedHeaders http.Header
+	writtenHeaders  bool
+	wsConn          *websocket.Conn
+	headers         http.Header
+	flushedHeaders  http.Header
+	timeOutInterval int
+	tickerCount     int
 }
 
 func newWebSocketResponseWriter(wsConn *websocket.Conn) *webSocketResponseWriter {
-	return &webSocketResponseWriter{
-		writtenHeaders: false,
-		headers:        make(http.Header),
-		flushedHeaders: make(http.Header),
-		wsConn:         wsConn,
+	wsRW := &webSocketResponseWriter{
+		writtenHeaders:  false,
+		headers:         make(http.Header),
+		flushedHeaders:  make(http.Header),
+		wsConn:          wsConn,
+		timeOutInterval: 30,
+		tickerCount:     0,
+	}
+	go wsRW.ping()
+	return wsRW
+}
+
+func (w *webSocketResponseWriter) ping() {
+	dispose := make(chan bool)
+	ticker := time.NewTicker(time.Second)
+	w.wsConn.SetCloseHandler(func(code int, text string) error {
+		close(dispose)
+		return nil
+	})
+	for {
+		select {
+		case <-dispose:
+			goto Stop
```

Comment:
Lets just return here.
```suggestion
			return
```